File: php/rest_api_return_top_hits.php
<?php

# This script makes a GET request to the /searches endpoint on the Analytics REST API - https://www.algolia.com/doc/rest-api/analytics/. 
# To get the top 1000 searches over the last 7 days.
# There is no API client for Analytics, so this script uses the PHP Requests library to make the call.

# Install the API client: https://www.algolia.com/doc/api-client/getting-started/install/php/?client=php
require __DIR__ . '/vendor/autoload.php';

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->load();

# Get your Algolia Application ID and (admin) API key from the dashboard: https://www.algolia.com/account/api-keys
# and choose a name for your index. Add these environment variables to a `.env` file:
$ALGOLIA_APP_ID = $_ENV['ALGOLIA_APP_ID'];
$ALGOLIA_API_KEY = $_ENV['ALGOLIA_API_KEY'];
$ALGOLIA_INDEX_NAME = $_ENV['ALGOLIA_INDEX_NAME'];

# The Analytics API can be reached from multiple domains, each specific to a region. 
# You should use the domain that matches the region where your analytics data is stored and processed. 
# View your analytics region: https://www.algolia.com/infra/analytics
# The following domains are available:
# United States: https://analytics.us.algolia.com
# Europe (Germany): https://analytics.de.algolia.com

$URL_DOMAIN = $_ENV['URL_DOMAIN'];

$url = "$URL_DOMAIN/2/searches?index=$ALGOLIA_INDEX_NAME&limit=1000";

# Describing HTTP request
$request_options  = [
    'http' => [
        'method'  => 'GET',
        'header'  => join("\r\n", [
            "X-Algolia-Application-Id: $ALGOLIA_APP_ID",
            "X-Algolia-API-Key: $ALGOLIA_API_KEY",
        ]),
    ]
];

# Sending HTTP request
$result = file_get_contents($url, false, stream_context_create($request_options));

if ($result === FALSE) {
    exit("Failed HTTP request");
}

# Returning 1000 Top Searches
print("1000 Top Searches:\n");
var_dump($result);

# Format the resulting json string
$json = json_decode($result);
$formatted_result = json_encode($json, JSON_PRETTY_PRINT);

# Write json to file
if (file_put_contents("{$ALGOLIA_INDEX_NAME}_top_1000_searches.json", $formatted_result))
    echo "JSON file created successfully...\n";
else
    echo "Oops! Error creating json file...\n";

File: go/rest_api_return_top_hits.go
package main

import (
	"fmt"
	"log"
	"os"
	"github.com/algolia/algoliasearch-client-go/v4/algolia/analytics"
	"github.com/joho/godotenv"
	"encoding/json"
)

func PrintErrAndExit(err error) {
	fmt.Println(err)
	os.Exit(1)
}

func main() {
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error loading .env file")
	}
	// Algolia client credentials
	appID, apiKey, indexName := os.Getenv("ALGOLIA_APP_ID"), os.Getenv("ALGOLIA_API_KEY"), os.Getenv("ALGOLIA_INDEX_NAME")

	// Start the API client
	// https://www.algolia.com/doc/libraries/sdk/methods/search#go
	// Initialize the client with your application region, eg. analytics.ALGOLIA_APPLICATION_REGION
	client, err := analytics.NewClient(appID, apiKey, analytics.US)
	if err != nil {
	// The client can fail to initialize if you pass an invalid parameter.
		fmt.Printf("Client error")
		panic(err)
	}

	resTopSearches, err := client.GetTopSearches(client.NewApiGetTopSearchesRequest(indexName).WithLimit(1000),)
	if err != nil {
	// handle the eventual error
		panic(err)
	}

	// Format the resulting json string
	resEncodeTopSearchesJson, err := json.Marshal(resTopSearches)
	if err != nil {
		log.Fatal(err)
	}

	// Write json to file
	err = os.WriteFile(indexName + "_top_1000_searches.json", resEncodeTopSearchesJson, 0644) // 0644 grants read/write permissions for the owner, and read for others
	if err != nil {
		log.Fatalf("Error writing to file: %v", err)
	}
}